### ISSUE TITLE
Matrix BETA | Sync is stuck - is anyone else having this issue???

### ISSUE THREAD

**Issue by Noustaa** [croneter/PlexKodiConnect#1283]
Hi,

Im on kodi Matrix beta 2 with PKC 3.0.3.
When i run the initial sync for all the plex content the sync is getting stuck after a while.
I'm attaching my log here : [kodi.log](https://github.com/croneter/PlexKodiConnect/files/5824945/kodi.log)

Let me know if you need anything else or if i can try anything in order to help you :)

Thanks, Noustaa

**Comment by croneter** [comment]
If you run the initial sync several times, will it always get stuck at the
same item?

**Comment by Noustaa** [comment]
Yes affirmative ! But the item is working fine on Plex i double checked

**Comment by croneter** [comment]
Please share that item's detailed XML metadata. See https://support.plex.tv/articles/202462186-viewing-item-details/, "Get Info". Seems to be the movie `Persona 3 : The Movie #2 - Midsummer Knight's Dream`

**Comment by Noustaa** [comment]
Hi thank you for your answer! 

NOTE: i have updated to Matrix RC1 today 

I am not owner of this PMS so do not have access to the XML. So i have tried with my own PMS to  be able to give you the XML but here it is stuck on a season item, not a movie item, so i guess there is no XML metadata for the season folder ? 

Here is the log: [kodi.log](https://github.com/croneter/PlexKodiConnect/files/5826295/kodi.log)

**Comment by croneter** [comment]
Yes there is a season xml. Kindly share that please. On your own PMS, does the sync also get stuck at the same item all the time? How many times did you test?

**Comment by Noustaa** [comment]
It looks like the target tvshow is not available anymore on the server but it is weird because a few time on my PMS the sync was completed successfully but 50% of the time it is stuck. i tried like 10 times, 5 times OK and 5 times stuck. On the other PMS that i do not own by the way it never succeeded it was always stuck.
As far as i know it is often on the same item but not always ! 

Here is the XML, i got it on the episode because on the season itself i can't find anything, hope it is correct (there is only 2 episodes so i will put both):
Episode 1: https://pastebin.com/ErRxsEYc
Episode 2: https://pastebin.com/WtPiF6Za

**Comment by croneter** [comment]
Hmm, apart from the language I don't see any possible issue. 

You CAN get an XML for a season: Simply open any XML in the browser, e.g. one you shared. In the URL, replace the number `xxxx` of the part `library/metadata/xxxx` with the Plex ID for the season: `461069`. Please double check whether it is indeed the correct season XML

**Comment by viseniv** [comment]
I am having the same issue. 
Mine stalls at 97% and not always on the same movie, if i delete the element on stucks on next start stalls at 97% again on another movie

**Comment by Noustaa** [comment]
Hi i'm sorry i didn't have time to test further with your last instructions Croneter! I will try to work on it in these days.

Viseniv thanks for your feedback ! Could you please follow the same steps that croneter asked me to do to give him all the needed info to sort that out ? 

+ the kodi logs as usual -> https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug

**Comment by croneter** [comment]
@viseniv and @Noustaa 
The bug is extremly difficult to hunt down. It's probably a racing condition. Could you kindly manually install this version of PlexKodiConnect (Python3 for Kodi Matrix!), make sure debug logging within Kodi is enabled, then try to reproduce the Sync getting stuck?
[plugin.video.plexkodiconnect.zip](https://github.com/croneter/PlexKodiConnect/files/5885977/plugin.video.plexkodiconnect.zip)

**Comment by Noustaa** [comment]
Hi Croneter!
Thank you for your support.
I tried a first time and it did not stuck but i have got a sync error at the end, here is the log and a video for the error.

video: https://user-images.githubusercontent.com/69060343/106132956-e9ad3a80-6164-11eb-8c39-93a169c78bf4.mp4
log: https://wetransfer.com/downloads/75ba0546283d49048725de92999faf4220210128113359/4ca5acd6fa2f0d11b9b43fa492b08b7b20210128113410/db44f9




I will re try several time (like 10 times maybe) to see if i can reproduce the stuck sync again.

**Comment by croneter** [comment]
Thanks a lot @Noustaa !

Hmm, I don't see why you got an error. Which is really weird! It seems that the PMS is struggling to send the requested metadata though, you got a ton of these entries:
```
DEBUG <general>: PLEX.plex_functions: Waiting for download to finish
```

Unfortunately, you'll have to try until you can replicate the stuck sync...

**Comment by croneter** [comment]
Ah OK, I think the error is because you quit Kodi while PKC was still synching playstates. Nothing to worry about at this point

**Comment by croneter** [comment]
@Noustaa If the sync does not get stuck while processing a movie library, I think it won't get stuck at all - so you could just quit Kodi and start again (fastest if you just delete all databases in the Kodi userdata directory with the exception of "MyAddonsXXX.db" once you quit Kodi).

**Comment by Noustaa** [comment]
@croneter Thanks for the tip it is indeed faster like that haha ! 
SO 2nd time it was just stuck but at the very last item of the sync
here is the log: https://filebin.net/vc8mffiicwubboro

I'm keepp trying to get the same stuck as last time

**Comment by croneter** [comment]
@Noustaa Very interesting log, thanks! Did you cancel the sync while the last movie was being synched? How long did you wait?

**Comment by croneter** [comment]
Another log wouldn't hurt 🙈

**Comment by Noustaa** [comment]
I just did ALT+F4 after 2 or 3  minutes...

I'm trying to reproduce and i even re tried with the PMS i do not own and no stuck here... 
I keep trying a few more times !

**Comment by Noustaa** [comment]
Well it is not stucking like before... I even tried again with the PMS that i do not own which was stucking every single time and no problem here ! Maybe you did fix the issue with the last release. My first test was with 3.0.3 and now it seems to be working fine so i dont know !

I will download 3.0.6 now and see if it works.

**Comment by Noustaa** [comment]
Well... working fine also with 3.0.6 ! 😮

**Comment by Noustaa** [comment]
Damn it... i found the problem ! It looks like it was me fault :( !

I did reproduce the stuck ! Here is the log:
[kodi.zip](https://github.com/croneter/PlexKodiConnect/files/5888125/kodi.zip)
At the end i did alt F4 to close kodi.

The problem is due to the fact that i was systematically changing these parameters 
![image](https://user-images.githubusercontent.com/69060343/106156250-985f7400-6181-11eb-8673-0fa44d697694.png)

Because i though the the sync would go faster.... With the default settings it seems to be working fine !

**Comment by croneter** [comment]
Nope, @Noustaa , your last log provided a clou! Hopefully this issue is fixed now with 3.0.7  / 2.12.12. Thanks a lot for reporting back!
